polyfactory/factories/odmantic_odm_factory.py

```diff
@@ -36,8 +35,6 @@ def is_supported_type(cls, value: Any) -> "TypeGuard[type[T]]":
         """
         return is_safe_subclass(value, Model) or is_safe_subclass(value, EmbeddedModel)
 
-    id = Ignore()
```

Comment:
Note regarding the bson.Regex below - the library has a regex factory. You don't need faker, see the handling of constrainted strings